src/python/WMCore/Storage/TrivialFileCatalog.py

```diff
@@ -196,55 +232,144 @@ def tfcFilename(contactString):
+                result = entry.get("result", None)
+                chain = entry.get("chain", None)
+                if True in (protocol, match == None):
+                    continue
+                tfcInstance.addMapping(str(protocol), str(match), str(result), chain, mapping)
+      #return tfcInstance
+    else:
+        try:
+            jsonFile = open(filename,encoding="utf-8")
+            jsElements = json.load(jsonFile)
+        except Exception as ex:
+            msg = "Error reading FileCatalog: %s\n" % filename
+            msg += str(ex)
+            raise RuntimeError(msg)
+        #now loop over elements, select the right one and fill lfn-to-pfn
+        for jsElement in jsElements:
+            if jsElement['site'] == storageAttr['site'] and jsElement['volume'] == storageAttr['volume']: 
+                #now loop over protocols
+                for proc in jsElement['protocols']:
+                    #check found match
+                    if proc['protocol'] == storageAttr['protocol']:
+                        chain = proc['chain'] if 'chain' in proc.keys() else None
+                        #check if prefix is in protocol block
+                        if 'prefix' in proc.keys():
+                            #lfn-to-pfn
+                            match = '(.*)' #match all
+                            result = proc['prefix']+'/$1'
+                            tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                            #pfn-to-lfn
+                            match = proc['prefix']+'/(.*)'
+                            result = '/$1'
+                            tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+                        #here is rules  
+                        else:
+                            #loop over rules
+                            for rule in proc['rules']:
+                                match = rule['lfn']
+                                result = rule['pfn']
+                                tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                                #pfn-to-lfn: not sure about this!!!
+                                match = rule['pfn'].replace('$1','(.*)')
+                                result = rule['lfn'].replace('/+','/').replace('^/','/')
+                                #now replace anything inside () with $1, for example (.*) --> $1, (store/.*) --> $1
+                                result = re.sub('\(.*\)','$1',result)
+                                tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+    
+    return tfcInstance
 
+def rseName(storageAttr):
+    rse = None
+    catalog = getCatalog(storageAttr)
+    storageJsonName = tfcFilename(catalog)
     try:
-        node = xmlFileToNode(filename)
+        jsonFile=open(storageJsonName,encoding="utf-8")
```

Comment:
Can you please replace it by a context manager? A construction like:
```
with open(storageJsonName, encoding="utf-8") as jsonFile:
  jsElements = json.load(jsonFile)
```
?

Reply:
Yes I will